components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -133,7 +152,12 @@ class GeckoEngineSession(
      * See [EngineSession.reload]
      */
     override fun reload(flags: LoadUrlFlags) {
-        geckoSession.reload(flags.value)
+        val loadRequest = pendingLoadRequest
+        if (loadRequest != null) {
+            loadUrl(loadRequest.url, loadRequest.parent, loadRequest.flags, loadRequest.additionalHeaders)
```

Comment:
Another question here is what would happen with a session that's persisted and restored in that state? Currently would be restored as a white tab and also can't be reloaded. Maybe handle that separately? Or should we persist a flag somehow and load again? 🤔

Reply:
Thinking through options here:
- We could not persist sessions that never had a completed load (never completed in either error of success)
- We could try and persist/restore this additional state somehow
- We could punt on that for now :)

Reply:
Looks maybe option 1 is best? File a follow-up to not persists tabs that never had a successful load?

Reply:
https://github.com/mozilla-mobile/android-components/issues/7859